include/rmm/thrust_rmm_allocator.h

```diff
@@ -58,12 +58,7 @@ class rmm_allocator : public thrust::device_malloc_allocator<T>
   
     inline void deallocate(pointer ptr, size_t)
     {
-      rmmError_t error = RMM_FREE(thrust::raw_pointer_cast(ptr), stream);
-  
-      if(error != RMM_SUCCESS)
-      {
-        throw thrust::system_error(error, thrust::cuda_category(), "rmm_allocator::deallocate(): RMM_FREE");
-      }
+      RMM_FREE(thrust::raw_pointer_cast(ptr), stream);
```

Comment:
Won't this now silently suppress a pending CUDA error?  `rmm:free` calls `cudaGetLastError` and squashes all CUDA error codes down to RMM_ERROR_CUDA_ERROR (not sure how one is supposed to detect what the specific CUDA error was).  This code ignores the return code from `rmm:free`, so the pending CUDA error gets cleared and silently dropped.

If we can't convey the error in a given code context it would be nice if we could leave it pending, but I don't know of a way to set a pending CUDA error once it's been cleared.  Arguably any pending error here is the fault of the prior code failing to check for it, but it would be nice to indicate somehow that something went wrong and couldn't be handled properly (e.g.: print to stderr).

Reply:
Yes, I've now removed the `cudaGetLastError()` from `rmm::free`.

Reply:
> Arguably any pending error here is the fault of the prior code failing to check for it, but it would be nice to indicate somehow that something went wrong and couldn't be handled properly

It shouldn't be the job of `rmm::free` to let you know that there is a pending CUDA error.

As far as deallocation errors go, we do print to `stderr` (in a DEBUG build) if it fails. This is handled down at the level of the memory resources, e.g., https://github.com/rapidsai/rmm/blob/branch-0.11/include/rmm/mr/cuda_memory_resource.hpp#L74.

Reply:
> It shouldn't be the job of rmm::free to let you know that there is a pending CUDA error.

Now that rmm:free doesn't clear errors, that makes sense to me.